STYLEGUIDE.adoc

```diff
@@ -34,6 +86,22 @@ In the past we tried `ormolu`, but its https://github.com/tweag/ormolu/issues/64
 
 Every top-level binding should have a type signature. _Most_ other named bindings (e.g. `let` and `where` bindings) should also have type signatures.
 
+Constraints should be uncurried.
+
+Bad:
+
+[source,Haskell]
+----
+foo :: Eq a => Ord a => a -> a -> a
+----
+
+Good:
+
+[source,Haskell]
+----
+foo :: (Eq a, Ord a) => a -> a -> a
+----
+
```

Comment:
I think this "curried" constraining has something to do with Haskell inferencing. I don't remember anymore about it, but there was a tweag's video about it sometime ago

Reply:
I mean the above (Bad) is not just the curried version of the below (Good). There is semantic difference in the type.

Reply:
Maybe it's this: https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/quantified_constraints.html?highlight=constraints

Reply:
> There is semantic difference in the type.

I'm not sure how the semantics are different, but the reason I prefer uncurried constraints is because it's easier to tell, in a big type signature, which are constraints and where the term arguments start.